InterGalacticCurrencyConvertor/src/main/java/statement/processor/StatementProcessor.java

```diff
@@ -0,0 +1,12 @@
+package main.java.statement.processor;
+
+import main.java.DataProcessResult;
+import main.java.ProcessorData;
+
+/**
+ * Contract that defines a statement.
+ */
+public interface StatementProcessor {
```

Comment:
Do you need a separate interface for QueryProcessor and StatementProcessor? They expose the same interface - you could just rename the processFoo(ProcessData) method to process(ProcessData).

Also, state the interface name with 'I'. e.g.

public interface IProcessor {
  boolean canProcess(ProcessorData data);
  DataProcessResult(ProcessorData data);
}

Reply:
done